src/realTimeReporter/utils.js

```diff
@@ -0,0 +1,8 @@
+const publishEvent = (event, msg) => process.emit(event, msg);
```

Comment:
This function incapsulates the event mechaism that we use.
If some external library will be used for this, it will be easy to change.
And such delegation helps to avoid Node.js warning about emmiting events.